File: TernaryOperator.cpp
// Ternary Operator in C++
// Rule: testCondition ? a : b 


#include <iostream>
using namespace std;



// Main function 
int main(int argc, char* argv[])
{

  int x,y,z;
  int m,n;

  x=1, y=2, z=4;


  m = x < y ? x : y;
  n = z < m ? z : m;
  cout << "x, y, z values are     : " << x << " " << y << " " << z << endl;
  cout << "Ternary operator result: " << n << endl;


  cout << endl;


  m = x > y ? x : y;
  n = z > m ? z : m;
  cout << "x, y, z values are     : " << x << " " << y << " "<< z << endl;
  cout << "Ternary operator result: " << n << endl;


  return 0;
}

File: TernaryOperator.java
// Ternary Operator in Java
// Rule: testCondition ? a : b 


public class TernaryOperator {


    public static void main(String[] args) {

	int x,y,z;
	int m,n;
	
	x=1; y=2; z=4;

	m = x < y ? x : y;
	n = z < m ? z : m;
	System.out.println("x, y, z values are : " + x + " " + y + " " + z);
	System.out.println("Ternary operator result: " + n);


	System.out.println("                     ");


	m = x > y ? x : y;
	n = z > m ? z : m;
	System.out.println("x, y, z values are : " + x + " " + y + " " + z);
	System.out.println("Ternary operator result: " + n);
    }

}
